server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -145,13 +150,19 @@ BytesReference getOrCompute(
         BytesReference cacheKey
     ) throws Exception {
         assert reader.getReaderCacheHelper() != null;
-        final Key key = new Key(cacheEntity, reader.getReaderCacheHelper().getKey(), cacheKey);
+        assert reader.getReaderCacheHelper() instanceof OpenSearchDirectoryReader.DelegatingCacheHelper;
```

Comment:
Are we assuming that the reader is always instanceof OpenSearchDirectoryReader? If yes, why we validating that explicitly in invalidate method?

Reply:
Yes, reader is always assumed to be instance of OpenSearchDirectoryReader. Regarding this check, this is more of a safeguard in place. As in case, if there are changes in how we wrap helper/reader, it should get caught.